docs/reference/components/sources/exec.cue

```diff
@@ -0,0 +1,231 @@
+package metadata
+
+components: sources: exec: {
+	title: "Exec"
+
+	classes: {
+		commonly_used: false
+		delivery:      "at_least_once"
+		deployment_roles: ["sidecar"]
+		development:   "beta"
+		egress_method: "stream"
+		stateful:      false
+	}
+
+	features: {
+		multiline: enabled: false
+		receive: {
+			from: {
+				service: services.exec
+			}
+
+			tls: enabled: false
+		}
+	}
+
+	support: {
+		targets: {
+			"aarch64-unknown-linux-gnu":      true
+			"aarch64-unknown-linux-musl":     true
+			"armv7-unknown-linux-gnueabihf":  true
+			"armv7-unknown-linux-musleabihf": true
+			"x86_64-apple-darwin":            true
+			"x86_64-pc-windows-msv":          true
+			"x86_64-unknown-linux-gnu":       true
+			"x86_64-unknown-linux-musl":      true
+		}
+		requirements: []
+		warnings: []
+		notices: []
+	}
+
+	installation: {
+		platform_name: null
+	}
+
+	configuration: {
+		command: {
+			required:    true
+			description: "The command to be run, plus any arguments required."
+			type: array: {
+				items: type: string: {
+					examples: ["echo", "Hello World!", "ls", "-la"]
+					syntax: "literal"
+				}
+			}
+		}
+		working_directory: {
+			common:      false
+			required:    false
+			description: "The directory in which to run the command."
+			warnings: []
+			type: string: {
+				default: null
+				syntax:  "literal"
+			}
+		}
+		include_stderr: {
+			common:      false
+			description: "Include the output of stderr when generating events."
+			required:    false
+			type: bool: default: false
+		}
+		event_per_line: {
+			common:      false
+			description: "Determine if events should be generated per line."
```

Comment:
I have modified the documentation text as you suggested. For a streaming process I think it is still possible to not want an `event_per_line` if your process outputs a continuous stream of bytes without a newline (e.g. `printf` without the `\n`). I agree that most of the time you would want `event_per_line` to be `true` though. I like the flexibility of being able to handle and with the default being true most of the time the user can ignore this config setting. Happy to move to only scheduled if you prefer though.

Reply:
Yeah, I guess I'm just wondering: if their stream has no newlines, how the events would otherwise be framed?

Reply:
Maybe fixed sized number of bytes maybe e.g. I tested with a script that used a `printf` to print hello every 3 seconds. I made the buffer size 5 and it worked perfectly. I do agree that this will be much less common though.

Reply:
Agreed, I think that will be fairly uncommon. We have been thinking about "codecs" recently though: https://github.com/timberio/vector/issues/6958. I think this could fit in with that work as a "fixed length" codec. Using the max buffer size to emulate that feels a bit indirect.

Reply:
Yes that sounds quite elegant! The tokio codecs are knew to me so I'll spend a bit of time trying to figure them out over the weekend.

Reply:
Changed to used the codec now. I really like it. I created a custom codec for bytes which we can remove and replace with the out of the box one if you want (I just liked the idea of applying the buffer to non-lined streamed output). I think if we go with the out of the box codec it will be of random size and the buffer size does not really apply.